### PR TITLE
Export code-tools jenkins scripts to Git SCM

### PR DESCRIPTION
**Pull request by M-Davies**
* Also fixes shellcheck problems within the scripts

Closes: #43 
Signed-off-by: Morgan Davies <morgandavies2020@gmail.com>

### CODE REVIEW

**Comment by M-Davies** [comment]
Testing at https://ci.adoptopenjdk.net/view/Dependencies/job/testing. So far I have the following jobs working:
- [x] https://ci.adoptopenjdk.net/view/Dependencies/job/testing/job/alpha-asmtools/11/console
- [ ] https://ci.adoptopenjdk.net/view/Dependencies/job/testing/job/alpha-jcov/
- [ ] https://ci.adoptopenjdk.net/view/Dependencies/job/testing/job/alpha-jtharness/
- [ ] https://ci.adoptopenjdk.net/view/Dependencies/job/testing/job/alpha-jtreg/
- [ ] https://ci.adoptopenjdk.net/view/Dependencies/job/testing/job/alpha-sigtest/

**Comment by sophia-guo** [comment]
I wonder if we should pull those tools from github instead of hg ( https://github.com/openjdk/  vs http://hg.openjdk.java.net/code-tools/) ? Looks like openjdk hg is stale all new changes only go to github?  For example: github jcov was updated 4 days ago while hg jcov was updated 11 months ago. For jcov the latest release requires asm8 and to work with jdk16 jcov requires asm9.1.  Using github repo building with latest release tag might work better with newer jdk version.

**Comment by karianna** [comment]
> I wonder if we should pull those tools from github instead of hg ( https://github.com/openjdk/ vs http://hg.openjdk.java.net/code-tools/) ? Looks like openjdk hg is stale all new changes only go to github? For example: github jcov was updated 4 days ago while hg jcov was updated 11 months ago. For jcov the latest release requires asm8 and to work with jdk16 jcov requires asm9.1. Using github repo building with latest release tag might work better with newer jdk version.

Yes we should swap to this for the nightlies - if they also have the tagged versions then we can go with GH for that as well.

**Comment by gdams** [comment]
Swapped to the skara repos. They have tags so all seems well.

**Comment by gdams** [comment]
@sophia-guo are you going to be able to change the test jobs to get the artefact from `./jtreg`? See https://ci.adoptopenjdk.net/job/testing/job/alpha_pipeline/32/artifact/ to see what the new layout would look like. Once you're happy we'll merge and create a new job

**Review by gdams — APPROVED** [review]

**Comment by gdams** [comment]
I think that the change needs to be made in https://github.com/AdoptOpenJDK/TKG/blob/master/scripts/getDependencies.pl but I'll let @sophia-guo confirm

**Comment by sophia-guo** [comment]
Yes @gdams . Both  https://github.com/AdoptOpenJDK/TKG/blob/master/scripts/getDependencies.pl and https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/openjdk/build.xml#L57 need to be updated. I will try to test it.

**Comment by sophia-guo** [comment]
Yes, that's good. I can update test materials in openjdk-tests and TKG.

File: tools/code-tools/asmtools.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+set -eu
+
+function generateArtifact() {
+    tag=$1
+    version=$2
+
+    echo "tag=$tag"
+    echo "version=$version"
+
+	artifact=asmtools-$version
```
Comment (sophia-guo) [inline tools/code-tools/asmtools.sh:12]: Couple of Indentation.

File: tools/code-tools/asmtools.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+	# In WORKSPACE until here
+	echo "Moving into build..."
+	cd build
+
+    perl -p -i -e 's/"9"/"1.8"/g' build.xml
+
+	export JAVA_TOOL_OPTIONS=-Dfile.encoding=UTF8
+
+	echo "Building asmtools"
```
Comment (sophia-guo) [inline tools/code-tools/asmtools.sh:27]: Indentation

**Review by sophia-guo — APPROVED** [review]
A couple of place have the indentation issue. Other than that, LGTM.

**Comment by gdams** [comment]
@M-Davies can you handle setting up the new pipeline?

**Comment by M-Davies** [comment]
> @M-Davies can you handle setting up the new pipeline?

Done. Test pipeline running https://ci.adoptopenjdk.net/view/Dependencies/job/dependency_pipeline/1/console

**Comment by gdams** [comment]
@sophia-guo The new pipeline is now available at https://ci.adoptopenjdk.net/view/Dependencies/job/dependency_pipeline/lastSuccessfulBuild/artifact/

Let me know when you've migrated the test jobs and I'll clear up the old jobs.

**Pull request merged**